File: download_mega.py
from mega import Mega
import sys

URL = "https://mega.nz/file/h7VmXBhC#jW1lH565j3ST9eeYLLcr6LjzOnOxX_Vx7FYpWFTJOEs"
DEST_DIR = "/workspace"

def main():
    mega = Mega()
    m = mega.login()  # anonymous
    out_path = m.download_url(URL, DEST_DIR)
    print(out_path)

if __name__ == "__main__":
    main()

File: download_mega.js
const fs = require('fs');
const path = require('path');
const { File } = require('megajs');

async function main() {
  const [,, url, destDir] = process.argv;
  if (!url || !destDir) {
    console.error('Usage: node download_mega.js <url> <dest_dir>');
    process.exit(1);
  }
  try {
    const file = File.fromURL(url);
    await file.loadAttributes();
    const outPath = path.join(destDir, file.name);
    await new Promise((resolve, reject) => {
      const writeStream = fs.createWriteStream(outPath);
      file.download().on('error', reject).pipe(writeStream);
      writeStream.on('finish', resolve);
      writeStream.on('error', reject);
    });
    console.log(outPath);
  } catch (err) {
    console.error('Download error:', err);
    process.exit(2);
  }
}

main();
